### ISSUE TITLE
Add support for HTTP basic auth?

### ISSUE THREAD

**Issue by AxelTerizaki** [Vahelnir/dlmanager#1]
Could it be possible ?

I'm trying to download a set of files behind a user/password basic auth Apache folder.

Thanks in advance.

**Comment by Vahelnir** [comment]
You can already do that !
You can use `http://username:password@yoururl`.
But i just added a way to push custom headers to a download.
```js
manager.addToQueue('http://yoururl/privateendpoint', 'foo/bar')
  .withHeaders({'Authorization': `Basic ${Buffer.from('username:password').toString('base64')}`})
```

**Comment by AxelTerizaki** [comment]
I did try that, but it didn't seem to work, but I believe I did a mistake now that I think again about it.

Passing custom headers would be cool though, it'd allow to use authorization tokens for instance :)

Sorry for opening an issue about it then.

**Comment by AxelTerizaki** [comment]
BTW I think the readme could need a little work, I had to look through the examples to get an idea of what to do sometimes. ^^

**Comment by Vahelnir** [comment]
Yeah, i know but tbh i didn't work on that lib since the last commit.
I need to refactor some of the code, add real comments and maybe use a tool like esdoc to create the doc automaticaly.
What do you think the readme lacks of ?

**Comment by AxelTerizaki** [comment]
This library is really good and could come in handy. I had to code my own one for now but I discovered your s a little late, so I'm still wondering if I should use it instead :p 

To me the doc lacks some finer examples, the states aren't detailed, and I had to look into the example to understand what the verification did as you say nowhere that addToQueue or download has a 3rd optional argument which is the md5sum of the downloaded file.

I don't think you need an autodocumentaiton as your class is fairly easy to understand if you look inside the code, but people will want more documentation I think :)

**Comment by Vahelnir** [comment]
I'll try to change that next week. 
Thanks for your feedback :)

**Comment by AxelTerizaki** [comment]
You're welcome :)

Not sure if I'll use yours as I customized mine, but looking at it for inspiration was interesting.

**Comment by Vahelnir** [comment]
you can always pr with some improvements or to let some more customization to the developer.
